### ISSUE TITLE
LanguageServer is extremely slow to initialize

### ISSUE THREAD

**Issue by randy3k** [JuliaEditorSupport/LanguageServer.jl#176]
I have been working on the package https://github.com/JuliaMatrices/BandedMatrices.jl
This package usually takes 2-3 minutes to precompile this package since it ships with [precompile.jl](https://github.com/JuliaMatrices/BandedMatrices.jl/blob/34d7871db2db261720e60e8a014333ab351439d0/src/precompile.jl) and it creates some issues when using LanguageServer.

Details:

If I made any changes to the package and restart the editor (Sublime Text in my case):

Firstly, LanguageServer initializes by linting all the files in the directory. Eventually, it lints [this file](https://github.com/JuliaMatrices/BandedMatrices.jl/blob/34d7871db2db261720e60e8a014333ab351439d0/docs/make.jl#L12-L12) which contains `using BandedMatrices`. Since some files have been changed, it will again takes 2-3 minutes to precompile the package. The LanguageServer is not responsive at all until the initialization has done.

I am not sure if there is any solution to this issue, but it has been an issue for me.

**Comment by davidanthoff** [comment]
Yes, this is kind of a known issue. We really, really need to move away from loading user packages into the language server process at all, not just for these delay issues but also to increase stability etc. @ZacLN and I have a plan that we came up with during juliacon, just the question when we find time to implement it :)

**Comment by non-Jedi** [comment]
As data, in one case (on my only slightly underpowered laptop), it took 16 minutes for me to have the LanguageServer ready for usage with a specific project (on first use of course).

**Comment by AlexLewandowski** [comment]
I think the start-up times are problematically long. As it stands, the language server is the main tool for "ide-like" support in Emacs. I can only imagine similar is true for other text editors. 

Having such a long initialization time can really put someone off from learning the language.

**Comment by HongLau** [comment]
Real data on my laptop with nvim + coc-julia:

- Initializing julia: 40 seconds
- Julia Language Server Indexing packages: 34 seonds

**Comment by einarpersson** [comment]
Is this something being worked on? As a new Julia user I have really liked much about the language so far, but the slow initialization (`starting julia language server` + `language server indexing packages`) is... off-putting. 

Is it supposed to be the case that whereas VSCode is ready in a few seconds, I have to wait minutes before I can start to actually code in Julia?

**Comment by davidanthoff** [comment]
Yes, we are making good progress on getting rid of indexing entirely.

**Comment by kataklasm** [comment]
As LS.jl is still painstakingly slow even on good machines, is this issue any further since the last update?

**Comment by jcbritobr** [comment]
Its taking a lot of time to indexing for me. With two big packages as CairoMakie and DataFrames. ~10 min on windows 10 x64, without an ssd.

**Comment by haxfn** [comment]
Why is the Julia language server written in Julia? Will the LanguageServer be any faster if written in any other language? Why is the LanguageServer so slow to load and also sometimes slow in going to definition (runtime).
